metricbeat/helper/prometheus/prometheus.go

```diff
@@ -165,6 +166,14 @@ func (p *prometheus) GetProcessedMetrics(mapping *MetricsMapping) ([]common.MapS
 
 				// value may be a mapstr (for histograms and summaries), do a deep update to avoid smashing existing fields
 				update := common.MapStr{}
+				if valueFloat64, ok := value.(float64); ok {
+					if math.IsNaN(valueFloat64) || math.IsInf(valueFloat64, 1) || math.IsInf(valueFloat64, 0) {
```

Comment:
`math.IsInf(value, 0)` checks if the value is `+Inf`  or `-Inf`.
```suggestion
					if math.IsNaN(valueFloat64) || math.IsInf(valueFloat64, 0) {
```

Reply:
ohh yes you are right, thanks!